lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -357,6 +357,32 @@ bool canFoldIntoConversion(Operation *op, Attribute targetEncoding) {
 
 //
 
+// Replace ForOp with a new ForOp with extra operands. The YieldOp is not
+// updated and needs to be updated separatly for the loop to be correct.
+scf::ForOp replaceForOpWithNewSignature(OpBuilder &rewriter, scf::ForOp loop,
```

Comment:
we should have a similar utility somewhere since the conversion optimization pass also generates loops with different signature (cc @ThomasRaoux )

Reply:
Right, Zahi moved this function from RemoveLayoutConversions.cpp so it is common